### ISSUE TITLE
JmDNS and Cellular Data issue

### ISSUE THREAD

**Issue by CobaCity** [jmdns/jmdns#116]
Hello. I have some problem using JmDNS on my Huawei Nexus 6P - Android 7.1 
When Wifi (connected to the printer) and Cellular Data is turned on , discovery does not work. 
If I turn off Cellular Data everything starts working.

**Comment by ViToni** [comment]
This issue is not very specific because it's not clear how you use JmDNS.

As you usually have to pass the InetAddress (by which the interface is selected) to JmDNS it is very probaly related to the selection of the interface / InetAddress.

**Comment by emcniece** [comment]
Thanks to @CobaCity I confirmed this behaviour in a Google Pixel and a Nexus 5x (both Android 8.0.0) - disabling mobile data seems to impact service discovery. Strangely, the OnePlus One that I just upgraded to LineageOS Android 8.0.0 works perfectly.

I'm testing this through 2 interfaces: [cordova-plugin-zeroconf](https://github.com/becvert/cordova-plugin-zeroconf) and a native Android app that I mashed together using some examples. Both use JmDNS, and both apps have the same results. I can post the Android app code if anyone is interested, but for now [here is the Cordova plugin code](https://github.com/becvert/cordova-plugin-zeroconf/blob/master/src/android/net/becvert/cordova/ZeroConf.java#L428-L437). Note that it is indeed binding to each available `InetAddress`.

**Comment by emcniece** [comment]
After much more testing I have confirmed that mobile data is indeed the source of the problem.  The OnePlus One on Android 8 exhibits this issue as well.

Network architecture is similar to the original description, but instead of a printer I am using a proprietary AP:

```
192.168.37.1                                   some IP
----------------------      ---------      ---------------
| AP w/ mdns service |______| Phone |______| Mobile Data |
| no internet        |      |       |      | w/ internet |
----------------------      ---------      ---------------
```

Until mobile data is disabled and the cell network IP goes away, I can't even ping or talk to the `192.168.37.1` host. As soon as mobile data is disabled, the Wifi AP works and I can receive mDNS scan results and open websockets again.

If the Wifi AP actually has internet access, scanning works fine. It's like the OS is forcing my app to use the network interface that has an internet connection. What really confuses me is that this behaviour exists even when a WifiManager is specified and locked... what do I care about the state of mobile data when I am only working with Wifi?! Crazy.

I have only experienced this in Android 8 - haven't tested at all in 7, and haven't had this issue at all in 6.

**Comment by eskim** [comment]
Same here.
JmDNS can't discover when both LTE and Wifi(no internet access) is enabled.

**Comment by ViToni** [comment]
@emcniece You might want to change `cordova-plugin-zeroconf` to use the JmDNS master branch since it includes the fix https://github.com/jmdns/jmdns/pull/150

**Comment by emcniece** [comment]
Thanks @ViToni, I'll be testing this over the next 4 days to confirm.

**Comment by emcniece** [comment]
Resolved by #150!! 🎉 

Tested on multiple Android 6/7/8 devices today 😀
